src/app/data-planes/components/DataPlaneList.vue

```diff
@@ -9,35 +9,42 @@
       { label: 'Service', key: 'service' },
       ...(!props.gateways ? [{ label: 'Protocol', key: 'protocol' }] : []),
       ...(isMultiZoneMode ? [{ label: 'Zone', key: 'zone' }] : []),
-      { label: 'Last Updated', key: 'lastUpdated' },
       { label: 'Certificate Info', key: 'certificate' },
       { label: 'Status', key: 'status' },
       { label: 'Warnings', key: 'warnings', hideLabel: true },
-      { label: 'Actions', key: 'actions', hideLabel: true },
+      { label: 'Details', key: 'details', hideLabel: true },
     ]"
     :page-number="props.pageNumber"
     :page-size="props.pageSize"
     :total="props.total"
     :items="props.items ? transformToTableData(props.items) : undefined"
     :error="props.error"
+    :is-selected-row="props.isSelectedRow"
     @change="emit('change', $event)"
   >
     <template #toolbar>
       <slot name="toolbar" />
     </template>
+
     <template #name="{ row: item }">
       <RouterLink
         :to="{
-          name: item.isGateway ? 'gateway-detail-view' : 'data-plane-detail-view',
+          name: props.summaryRouteName,
           params: {
+            mesh: item.mesh,
             dataPlane: item.name,
           },
+          query: {
+            page: props.pageNumber,
+            size: props.pageSize,
+          },
         }"
         data-testid="detail-view-link"
       >
         {{ item.name }}
       </RouterLink>
     </template>
+
```

Comment:
I still wish that we didn't have this reused `DataplaneList` component and instead just used AppCollection in their respective views. Looks like we are starting to add more props and complicate things, and moreover then use one approach in one place and a different approach in another:

- Why is there a property to specify the summary view route, but not one to specify the detail view route?
- Why does the detail route link use the `isGateway` property whereas the summary one doesn't?

Is there a reason why we use a different approach to render each link?

---

I also noticed that the data-testid here reads 'detail-view-link'. If it was me I'd probably change this to `data-testid="name"` or something. I also just noticed we've added `detail-link` a bit further down. Kinda not sure without fiddling myself but feels like we should name the things here a little differently/generically for when things change.

Just for clarity, I think we should at least change it so this link to the summary view isn't labelled as linking to the detail view. The larger "lets name things more generically for when they change" thing can wait and be done outside of this PR if that fits better with you.

Also just to note that this last point is very related to another conversation re: generic naming of properties elsewhere.

Reply:
> * Why is there a property to specify the summary view route, but not one to specify the detail view route?

Because the summary route is needed (because it cannot be determined purely based on whether the entry is a gateway or regular DPP as there are three relevant summary routes at play here) and the detail route isn’t.

> * Why does the detail route link use the `isGateway` property whereas the summary one doesn't?

See previous answer. It doesn’t need to.



> I also noticed that the data-testid here reads 'detail-view-link'. If it was me I'd probably change this to `data-testid="name"` or something. I also just noticed we've added `detail-link` a bit further down. Kinda not sure without fiddling myself but feels like we should name the things here a little differently/generically for when things change.

It’s pretty much the key information that those are detail links because that informs interaction rules under test.

> Just for clarity, I think we should at least change it so this link to the summary view isn't labelled as linking to the detail view. 

Good point. I removed this attribute because it’s not used anywhere.

Reply:
> But that link is no longer a link to the detail page, it changed to a link to a summary page i.e. the key information that those are detail links is now incorrect key information.

I’m of course referring to the link that does in fact go to the detail page and not the link that now goes to the summary view on which I have removed the attribute.

Reply:
> Would it have been possible to just reuse the same summary route for all three, or at least two? Are the summaries all summaries of a dataplane?

I think that’s possible, yes. I’ve added a commit for this.

While doing so I noticed that `RouteView` should probably not expect a `name` prop but just use `route.name` internally. Out of scope of course.

What I don’t think we can get rid of is the `props.summaryRouteName` from DataplaneList.

Reply:
> While doing so I noticed that RouteView should probably not expect a name prop but just use route.name internally. Out of scope of course.

From what I remember for a good while back now, I think that would give you the name of the route you are on, not the name of the route for the file itself.

Reply:
Oh right, yeah that makes perfect sense. It’s kinda annoying we have to "repeat" the route name like that. Easy to mess up and prevents reusability (at least if we intend to use this for something more specific).